core/provisioning-java/src/main/java/org/apache/syncope/core/provisioning/java/data/AuditDataBinderImpl.java

```diff
@@ -1,19 +1,60 @@
 package org.apache.syncope.core.provisioning.java.data;
 
-import org.apache.syncope.common.lib.to.AuditTO;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.builder.ToStringBuilder;
+import org.apache.commons.lang3.builder.ToStringStyle;
+import org.apache.syncope.common.lib.to.AuditEntryTO;
 import org.apache.syncope.core.persistence.api.entity.AuditEntry;
 import org.apache.syncope.core.provisioning.api.data.AuditDataBinder;
 import org.springframework.stereotype.Component;
 
+import java.util.Arrays;
+import java.util.stream.Collectors;
+
 @Component
 public class AuditDataBinderImpl implements AuditDataBinder {
+
     @Override
-    public AuditEntry create(final AuditTO applicationTO) {
-        return null;
+    public AuditEntryTO getAuditTO(final AuditEntry auditEntry) {
+        AuditEntryTO auditTO = new AuditEntryTO();
+        auditTO.setKey(auditEntry.getKey());
+        auditTO.setWho(auditEntry.getWho());
+        auditTO.setDate(auditEntry.getDate());
+        auditTO.setThrowable(auditEntry.getThrowable());
+        auditTO.setLoggerName(auditEntry.getLogger().toLoggerName());
+
+        if (StringUtils.isNotBlank(auditEntry.getLogger().getSubcategory())) {
```

Comment:
What's the purpose of such `if`? If `auditEntry.getLogger().getSubcategory()` is `null`, then no troubles in setting the newly created `audiTO`'s `subCategory` as `null` too.

Reply:
Cleaned up.